src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -1181,6 +1183,7 @@ def filter(self, col: Expression) -> "SelectStatement":
             new = SelectStatement(
                 from_=self.to_subqueryable(), where=col, analyzer=self.analyzer
             )
+        new._attributes = self._attributes
```

Comment:
flag control

Reply:
maybe we don't need this one because if the parameter is off, _attributes is always None. But we can also add

Reply:
you can have parameter turned off during middle. Let's protect the code for extra safety, and when we do copy, don't we also need to copy the attribute over?

Reply:
sure, let me add the parameter protection. I'd prefer not to copy the attributes for now, because `select()` use copy to create a new select_statement, where attributes may not be the same https://github.com/snowflakedb/snowpark-python/blob/016b063f4f95c0604d9572be5e2f7a68b437a4c7/src/snowflake/snowpark/_internal/analyzer/select_statement.py#L1114. Rather copying attributes then resetting attributes outside of copy, not copying it for now is safer